### PR TITLE
Fix sequential edit evaluation alignment and clarify locality metric protocols

### PR DESCRIPTION
**Pull request by BY-Elysia**
## Summary

This PR addresses two evaluation issues.

### 1. Fix sequential editing post-evaluation alignment

Previously, sequential editing could evaluate all requests using the final edited model state. This made `post` metrics misaligned: request 0, request 1, etc. were not necessarily evaluated against the model state immediately after their own edit.

This PR updates sequential evaluation so each request is evaluated right after its corresponding edit is applied.

### 2. Clarify locality metric protocols with `metric_meta`

`locality_acc` is not computed with one universal protocol across EasyEdit. Depending on the entry point, it can mean pre/post argmax token match, generation token match, full logits exact match, or top-k token match.

This PR adds lightweight `metric_meta` entries for locality metrics so downstream users can see how each metric was computed without changing existing metric values.

Each `metric_meta` entry contains:

```json
{
  "result_key": "the concrete metric field this metadata describes",
  "protocol": "the evaluation protocol used to produce the metric",
  "scorer": "the scoring rule used by that protocol",
  "comparable_group": "the group of metrics that are directly comparable"
}

### CODE REVIEW

**Comment by BY-Elysia** [comment]
Updated ROME/R-ROME layer statistics dataset loading to use current Hugging Face dataset identifiers. The old aliases `wikitext` and `wikipedia` are replaced with explicit dataset IDs/configs (`Salesforce/wikitext` and `wikimedia/wikipedia`), and `wikitext2` is added as a supported option. This keeps the stats collection utility compatible with current HF datasets while preserving the existing `wikitext` and `wikipedia` CLI options. Static compile checks and `git diff --check` passed.
